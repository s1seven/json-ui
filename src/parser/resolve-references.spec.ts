import { JSONSchema7 } from "json-schema";
import $RefParser from "@apidevtools/json-schema-ref-parser";

test("it matches the snapshot", async () => {
  const schema = (await import(
    "../../fixtures/schema-with-ref.json"
  )) as unknown as JSONSchema7;
  const result = await $RefParser.dereference(schema);
  expect(result).toMatchSnapshot();
});
