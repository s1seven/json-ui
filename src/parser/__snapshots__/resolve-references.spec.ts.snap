// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "default": {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": {
      "address": {
        "properties": {
          "city": {
            "type": "string",
          },
          "state": {
            "type": "string",
          },
          "street": {
            "type": "string",
          },
          "zip": {
            "type": "string",
          },
        },
        "required": [
          "street",
          "city",
          "state",
          "zip",
        ],
        "type": "object",
      },
    },
    "properties": {
      "user": {
        "properties": {
          "address": {
            "properties": {
              "city": {
                "type": "string",
              },
              "state": {
                "type": "string",
              },
              "street": {
                "type": "string",
              },
              "zip": {
                "type": "string",
              },
            },
            "required": [
              "street",
              "city",
              "state",
              "zip",
            ],
            "type": "object",
          },
          "name": {
            "type": "string",
          },
        },
        "required": [
          "name",
          "address",
        ],
        "type": "object",
      },
    },
    "type": "object",
  },
  "definitions": {
    "address": {
      "properties": {
        "city": {
          "type": "string",
        },
        "state": {
          "type": "string",
        },
        "street": {
          "type": "string",
        },
        "zip": {
          "type": "string",
        },
      },
      "required": [
        "street",
        "city",
        "state",
        "zip",
      ],
      "type": "object",
    },
  },
  "properties": {
    "user": {
      "properties": {
        "address": {
          "properties": {
            "city": {
              "type": "string",
            },
            "state": {
              "type": "string",
            },
            "street": {
              "type": "string",
            },
            "zip": {
              "type": "string",
            },
          },
          "required": [
            "street",
            "city",
            "state",
            "zip",
          ],
          "type": "object",
        },
        "name": {
          "type": "string",
        },
      },
      "required": [
        "name",
        "address",
      ],
      "type": "object",
    },
  },
  "type": "object",
}
`;
